### PR TITLE
init ddns-go 5.3.6

### PR DESCRIPTION
**Pull request by sirpdboy**
Maintainer: me / @\<github-user> (find it by checking history of the package Makefile)
Compile tested: (put here arch, model, OpenWrt version)
Run tested: (put here arch, model, OpenWrt version, tests done)

Description:

### CODE REVIEW

File: net/ddns-go/Makefile [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+PKG_NAME:=ddns-go
+PKG_VERSION:=5.3.6
+PKG_RELEASE:=1
+
+PKG_SOURCE_PROTO:=git
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=https://github.com/jeessy2/ddns-go.git
+PKG_MIRROR_HASH:=7b865a8b9618147514c4bb6220f02d2404ab5f6586e27797b717b4d7b61b81b6
+PKG_SOURCE_VERSION:=4d33865890bffebeebb1fe49f079d67c1d8c4625
```
Comment (jefferyto) [inline net/ddns-go/Makefile:18]: Use codeload or checkout the tag (`PKG_SOURCE_VERSION:=v$(PKG_VERSION)`.

File: net/ddns-go/Makefile [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+PKG_LICENSE:=AGPL-3.0-only
```
Comment (jefferyto) [inline net/ddns-go/Makefile:20]:
```suggestion
PKG_LICENSE:=MIT
```

See [LICENSE](https://github.com/jeessy2/ddns-go/blob/v5.3.6/LICENSE).

File: net/ddns-go/Makefile [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+PKG_CONFIG_DEPENDS:= \
+	CONFIG_DDNS-GO_COMPRESS_GOPROXY \
+	CONFIG_DDNS-GO_COMPRESS_UPX
+
+PKG_BUILD_DEPENDS:=golang/host
+PKG_BUILD_PARALLEL:=1
+PKG_USE_MIPS16:=0
+
+GO_PKG:=github.com/jeessy2/ddns-go/v5
+GO_PKG_LDFLAGS:=-s -w
```
Comment (jefferyto) [inline net/ddns-go/Makefile:33]: Setting `GO_PKG_LDFLAGS:=-s -w` is not necessary.

File: net/ddns-go/Makefile [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+GO_PKG_LDFLAGS_X:=m.version=$(PKG_VERSION)
+
+include $(INCLUDE_DIR)/package.mk
+include $(TOPDIR)/feeds/packages/lang/golang/golang-package.mk
```
Comment (jefferyto) [inline net/ddns-go/Makefile:37]:
```suggestion
include ../../lang/golang/golang-package.mk
```

File: net/ddns-go/Makefile [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+define Package/ddns-go
+  TITLE:=A Linux web GUI client of ddns-go
```
Comment (jefferyto) [inline net/ddns-go/Makefile:40]: I don't think this package is the GUI client.

File: net/ddns-go/Makefile [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+  SECTION:=net
+  CATEGORY:=Network
+  SUBMENU:=Web Servers/Proxies
```
Comment (jefferyto) [inline net/ddns-go/Makefile:43]:
```suggestion
  SUBMENU:=IP Addresses and Names
```

See https://github.com/openwrt/packages/blob/e88b7d25a2a4232d24574fb2501ff6886a3efabb/net/ddns-scripts/Makefile#L20

File: net/ddns-go/Makefile [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+config DDNS-GO_COMPRESS_UPX
+	bool "Compress executable files with UPX"
+	depends on !mips64
+	default n
+endef
+
+ifeq ($(CONFIG_DDNS-GO_COMPRESS_GOPROXY),y)
+	export GO111MODULE=on
+	export GOPROXY=https://goproxy.baidu.com
+endif
```
Comment (jefferyto) [inline net/ddns-go/Makefile:67]: We are not going to hard-code a goproxy in the Makefile.

File: net/ddns-go/Makefile [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+if [ -z "$${IPKG_INSTROOT}" ]; then
+	[ -f /etc/uci-defaults/luci-ddns-go ] && /etc/uci-defaults/luci-ddns-go && rm -f /etc/uci-defaults/luci-ddns-go
+fi
+endef
+
+define Build/Compile
+	$(call GoPackage/Build/Compile)
+ifneq ($(CONFIG_DDNS-GO_COMPRESS_UPX),)
+	$(STAGING_DIR_HOST)/bin/upx --lzma --best $(GO_PKG_BUILD_BIN_DIR)/ddns-go
+endif
```
Comment (jefferyto) [inline net/ddns-go/Makefile:80]: 1. There is no upx in `$(STAGING_DIR_HOST)/bin`.
2. OpenWrt uses compressed file systems, upx does not help - see #8000.

File: net/ddns-go/Makefile [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+endef
+
+define Package/$(PKG_NAME)/install
+	$(call GoPackage/Package/Install/Bin,$(1))
+
+	$(INSTALL_DIR) $(1)/etc/init.d
+	$(INSTALL_BIN) ./file/ddns-go.init $(1)/etc/init.d/ddns-go
+
+	$(INSTALL_DIR) $(1)/etc/uci-defaults
+	$(INSTALL_BIN) ./file/luci-ddns-go.uci-default $(1)/etc/uci-defaults/luci-ddns-go
```
Comment (jefferyto) [inline net/ddns-go/Makefile:90]: Services are already started by default on install, see https://github.com/openwrt/openwrt/blob/e0fb38f4eeed6cd56d4b5178aeb91442b4fe92a5/package/base-files/files/lib/functions.sh#L300-L303

File: net/ddns-go/file/ddns-go.init [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+
+START=99
+USE_PROCD=1
+
+PROG=/usr/bin/ddns-go
+CONFDIR=/etc/ddns-go
+CONF=/etc/ddns-go/ddns-go-config.yaml
+
+get_config() {
+        config_get_bool enabled $1 enabled 1
```
Comment (jefferyto) [inline net/ddns-go/file/ddns-go.init:19]: I don't think `enabled` should default to true, this lets the service start before the user has had a chance to configure it.

File: net/ddns-go/file/ddns-go.init [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+# This is free software, licensed under the Apache License, Version 2.0 .
+#
+
+
+START=99
+USE_PROCD=1
+
+PROG=/usr/bin/ddns-go
+CONFDIR=/etc/ddns-go
+CONF=/etc/ddns-go/ddns-go-config.yaml
```
Comment (jefferyto) [inline net/ddns-go/file/ddns-go.init:16]: Perhaps the package should have a `conffiles` section to save this file / directory.

File: net/ddns-go/file/ddns-go.init [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+	[ -d $CONFDIR ] || mkdir -p $CONFDIR 2>/dev/null
+	cat /usr/share/ddns-go/ddns-go-default.yaml  > $CONF
+}
+
+stop_service() {
+	ddns-go_prepare
+}
+
+start_service() {
+	ddns-go_prepare
```
Comment (jefferyto) [inline net/ddns-go/file/ddns-go.init:38]: Why do you need to kill the service before you start it?

File: net/ddns-go/file/ddns-go.init [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+	logger -t ddns-go -p warn "ddns-go is stop."
+}
+
+init_yaml(){
+	[ -d $CONFDIR ] || mkdir -p $CONFDIR 2>/dev/null
+	cat /usr/share/ddns-go/ddns-go-default.yaml  > $CONF
+}
+
+stop_service() {
+	ddns-go_prepare
```
Comment (jefferyto) [inline net/ddns-go/file/ddns-go.init:34]: If you are using procd to manage the service, why do you need to manually kill it?

**Review by jefferyto — COMMENTED** [review]
Also please read the [contributing guidelines](https://github.com/openwrt/packages/blob/master/CONTRIBUTING.md#commits-in-your-pull-requests-should). In particular, a real name is required for sign-offs.

**Comment by 1715173329** [comment]
https://github.com/immortalwrt/packages/tree/master/net/ddns-go

I have a "clean" version of this package. Anyway I think ddns-script is powerful enough so I didn't add it here.
(I have never actually used ddns-go)

**Comment by 1715173329** [comment]
Closed due to no response.

**Pull request closed without merging**